src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -127,9 +129,10 @@ private void startSearchThread(SourceTextEntry newEntry, boolean force) {
         UIThreadsUtil.mustBeSwingThread();
 
         clear();
+        List<GlossaryEntry> glossaryTerms = Core.getGlossaryManager().searchSourceMatches(newEntry);
```

Comment:
Thanks for your patience on this. This is definitely what I asked for... but now I'm really curious about how expensive this call is, since it's happening on the UI thread and I don't see a nice way to move it off without duplicating work or doing some complex synchronization.

If you have time, could you time it e.g. with `System.currentTimeMillis()` and make sure it's not crazy?

Reply:
Tested on a project with 1386 glossary terms. First run was 92 ms, all subsequent runs were 1-3 ms. CPU is Core i5-6200U@2.3GHz.

Also I noticed that `startSearchThread` is called twice for each segment: first time when I switch to a new segment, second time when I press Ctrl+M (automatic MT translation is disabled).